### ISSUE TITLE
The ‘balloon-eval’ doesn't work properly on Windows ...

### ISSUE THREAD

**Issue by cs86661** [vim/vim#1512]
Environment:
  Windows 7 64-bit SP1

Command:
  gvim.exe -u NONE -U NONE --noplugin -c "set balloondelay=100 balloonexpr=v:beval_text ballooneval" ..\\_vimrc

The balloon tip just can show a few times, and then it doesn't work any more ...

**Comment by chrisbra** [comment]
can you be a bit more precise? What exactly do you have to do to make the balloon disappear?

**Comment by cs86661** [comment]
I use balloon tip to translate english, but I found it only show only 1~3 times, and then don't show any more.
The command "balloonexpr=v:beval_text" is just a test example. You can move the mouse point to any word to make it show tip.

**Comment by cs86661** [comment]
I also confirm this issue on Windows 10 64-bit (GVIM version 8.0.69)
You can change to balloonexpr=expand('%'), the function "expand('%')" return current buffer name.
Then move mouse point to any words, see it show up the tip, may be 1~n times will not show again ...

Can anyone also confirm this ? :(

**Comment by niva-xx-zz** [comment]
Hi,

I confirm it works on gvim issue 8.0.427 on windows 10 64bit compiled with this example:
`
" Returns either the contents of a fold or spelling suggestions.
function! MyBalloonExpr()
  let foldStart = foldclosed(v:beval_lnum )
  let foldEnd = foldclosedend(v:beval_lnum)
  let lines = []
  if foldStart < 0
    " We're not in a fold.
    " If 'spell' is on and the word pointed to is incorrectly spelled,
    " the tool tip will contain a few suggestions.
    let lines = spellsuggest( spellbadword( v:beval_text )[ 0 ], 5, 0 )
  else
    let numLines = foldEnd - foldStart + 1
    " Up to 31 lines get shown okay; beyond that, only 30 lines are shown with
    " ellipsis in between to indicate too much. The reason why 31 get shown ok
    " is that 30 lines plus one of ellipsis is 31 anyway.
    if ( numLines > 31 )
      let lines = getline( foldStart, foldStart + 14 )
      let lines += [ '-- Snipped ' . ( numLines - 30 ) . ' lines --' ]
      let lines += getline( foldEnd - 14, foldEnd )
    else
      let lines = getline( foldStart, foldEnd )
    endif
  endif
  return join( lines, has( "balloon_multiline" ) ? "\n" : " " )
endfunction

set bexpr=MyBalloonExpr()
set ballooneval

`

**Comment by k-takata** [comment]
Maybe fixed by [8.1.0652](https://github.com/vim/vim/commit/6d9e71ad994fd428ecea76741376807cd18308ec)?

**Comment by cs86661** [comment]
> Maybe fixed by [8.1.0652](https://github.com/vim/vim/commit/6d9e71ad994fd428ecea76741376807cd18308ec)?

I tested 8.1.0652 and nothing different …… :(

**Comment by cs86661** [comment]
> Maybe fixed by [8.1.0652](https://github.com/vim/vim/commit/6d9e71ad994fd428ecea76741376807cd18308ec)?

I tested 8.1.0652 and nothing different …… :(

**Comment by cs86661** [comment]
> Maybe fixed by [8.1.0652](https://github.com/vim/vim/commit/6d9e71ad994fd428ecea76741376807cd18308ec)?

I tested 8.1.0652 and nothing different …… :(

**Comment by chrisbra** [comment]
possibly related: Patch [8.1.1300](2f10658b06bbdd8f25c4ff15226)

**Comment by cs86661** [comment]
> possibly related: Patch [8.1.1300](2f10658b06bbdd8f25c4ff15226)

I tested 8.1.1328 and still nothing different …… :(

**Comment by cs86661** [comment]
> possibly related: Patch [8.1.1300](2f10658b06bbdd8f25c4ff15226)

YA! I found I can input `call balloon_show('')` to temporary fix it!

**Comment by char101** [comment]
In my case `WM_MOUSEMOVE` is sent event every +/- 100ms  with the same coordinate thus causing the balloon not to appear because the idle detection is not working. It can be fixed by checking the mouse coordinate in `TrackUserActivity` in `gui_w32.c`.

```c
static POINT s_mousemove;

 static void
TrackUserActivity(UINT uMsg)
{
    if (uMsg == WM_MOUSEMOVE) {
	POINT p;
	GetCursorPos(&p);
	if (p.x == s_mousemove.x && p.y == s_mousemove.y) {
	    return;
	}
	s_mousemove.x = p.x;
	s_mousemove.y = p.y;
    }

    if ((uMsg >= WM_MOUSEFIRST && uMsg <= WM_MOUSELAST)
	    || (uMsg >= WM_KEYFIRST && uMsg <= WM_KEYLAST))
	LastActivity = GetTickCount();
}
```

**Comment by brammool** [comment]
There is also code for WM_MOUSEMOVE in HandleMouseHide().  Can they be combine?

**Comment by char101** [comment]
I don't think so since `HandleMouseHide` is called in a lot of places, but we can extract the logic to detect mouse movement out and then use it in both `HandleMouseHide` and `TrackUserActivity`.
